docs/run-tests-on-local.md

```diff
@@ -0,0 +1,188 @@
+    ```
+
+
+## For TAF users
+`Run tests either edgeX-taf-common Container or python comman`
+### Using The edgeX-taf-common Container To Run Testing
+####  Variables configuration
+Export the following variables that depend on running environment
+``` 
+# Required variables
+export WORK_DIR=${HOME}/edgex-taf
+```
+
+#### Run test by shell script with arguments
+```
+cd ${WORK_DIR}/TAF/utils/scripts/docker
+sh run-tests.sh ${ARCH} ${SECURITY_SERVICE_NEEDED} ${TEST_STRATEGY}
+# ex. sh run-tests.sh x86_64 false 1
+
+# Variable values
+${ARCH}: x86_64 | arm64
+${SECURITY_SERVICE_NEEDED}: false | true
+${TEST_STRATEGY}: 1 (functional) | 2 (integration)
+```
+
+#### Test report
+Open the report file by browser: ${WORK_DIR}/TAF/testArtifacts/reports/cp-edgex/v2-api-test.html
+
+
+
+### Using Python Command To Run Testing
+#### Setup required library
+1. Install pre-request packages:
+    Download pip3 and run this command:
+    ``` bash
+    sudo apt-get install python3-pip
+    ```
+2. Install TAF common:
+    ``` bash
+    cd ${HOME}/edgex-taf
+    git clone https://github.com/edgexfoundry/edgex-taf-common.git
+    
+    # Install dependency lib
+    pip3 install -r ./edgex-taf-common/requirements.txt
+
+    # Install edgex-taf-common as lib
+    pip3 install ./edgex-taf-common
+    ```
+3. Prepare test environment:
+    ``` bash
+    # Fetch the latest docker-compose file
+    cd ${HOME}/edgex-taf/TAF/utils/scripts/docker
+    sh get-compose-file.sh ${USE_DB} ${ARCH} ${USE_SECURITY}
+    # ex. sh get-compose-file.sh -redis x86_64 -
+    
+    # Variables for get-compose-file.sh
+    ${USE_DB}: -redis | -mongo (mongo is not supported from hanoi release)
+    ${ARCH}: x86_64 | arm64
+    ${USE_SECURITY}: - (false) | -security- (true)
+
+    # export the environment variables that depend on your environment.
+    export ARCH=x86_64
+    export SECURITY_SERVICE_NEEDED=false
+    export COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose:latest
+    ```
+#### Run Tests
+Run tests under ${HOME}/edgex-taf
+1. Deploy edgex:
+    ``` bash
+    # This step may take for a while if the edgex images don't exist on the machine
+    python3 -m TUC --exclude Skipped --include deploy-base-service -u deploy.robot -p default
+    ```
+2. Run Test
+    ###### Run V2 API Functional testing:
+    ``` bash
+    python3 -m TUC --exclude Skipped --include v2-api -u functionalTest/V2-API -p device-virtual
+    ```
+    ###### Run Integration testing:
+    ``` bash
+    python3 -m TUC --exclude Skipped -u integrationTest -p device-virtual
+    ```
+   
+3. Open the Test Reports
+   ###### Open the test reports in the browser. For example, to open the testing report, enter the following URL in the browser:
+    ``` bash
+    ${HOME}/edgex-taf/TAF/testArtifacts/reports/edgex/log.html
+    ```
+4. Shutdown edgex:
+    ``` bash
+    python3 -m TUC --exclude Skipped --include shutdown-edgex -u shutdown.robot -p default
```

Comment:
shutdown also needed sudo.  And I got an error when running this (and the containers did not come down).

```
jim@jim-ubuntu:~/forks/edgex-taf$ sudo python3 -m TUC --exclude Skipped --include shutdown-edgex -u shutdown.robot -p default
INFO:root:Run testing for profile 'default'
INFO:root:Running use case None
==============================================================================
default :: Shutdown EdgeX                                                     
==============================================================================
INFO:EdgeX shutdown:**************************************************
INFO:EdgeX shutdown:**************************************************
INFO:EdgeX shutdown:************  DEMO TEST RUN START  ************
INFO:EdgeX shutdown:Suite Setup: Logging started, level: INFO
INFO:EdgeX shutdown:Suite Setup: python version: 3.6.9 (default, Jan 26 2021, 15:33:00) 
[GCC 8.4.0]
INFO:EdgeX shutdown:Suite Setup: testcases version: 1.0
Shutdown EdgeX                                                        INFO:EdgeX shutdown:Shutdown all services
INFO:EdgeX shutdown:unknown shorthand flag: 'f' in -f

INFO:EdgeX shutdown:See 'docker run --help'.

INFO:EdgeX shutdown:exit 125
ERROR:EdgeX shutdown:Fail to execute cmd: sh /home/jim/forks/edgex-taf/TAF/utils/scripts/docker/shutdown.sh
[ ERROR ] Fail to execute cmd: sh /home/jim/forks/edgex-taf/TAF/utils/scripts/docker/shutdown.sh
Shutdown EdgeX                                                        | FAIL |
Fail to execute cmd: sh /home/jim/forks/edgex-taf/TAF/utils/scripts/docker/shutdown.sh
------------------------------------------------------------------------------
INFO:EdgeX shutdown:Suite teardown
INFO:EdgeX shutdown:************    DEMO TEST RUN END  ************
INFO:EdgeX shutdown:**************************************************
INFO:EdgeX shutdown:**************************************************
default :: Shutdown EdgeX                                             | FAIL |
1 critical test, 0 passed, 1 failed
1 test total, 0 passed, 1 failed
==============================================================================
Output:  /home/jim/forks/edgex-taf/TAF/testArtifacts/reports/edgex/report.xml
Log:     /home/jim/forks/edgex-taf/TAF/testArtifacts/reports/edgex/log.html
Report:  /home/jim/forks/edgex-taf/TAF/testArtifacts/reports/edgex/report.html

```

Reply:
I hit this error when the COMPOSE_IMAGE variable didn't export. Did you run this command in the same terminal that the variable was exported?

Reply:
Good catch Cherry.  I did was setting and unsetting the vars in various places to test it out and I might have missed this one.